dlt/normalize/items_normalizers.py

```diff
@@ -579,6 +583,9 @@ def _write_with_dlt_columns(
             for batch in pyarrow.pq_stream_with_new_columns(
                 f, new_columns, row_groups_per_read=self.REWRITE_ROW_GROUPS
             ):
+                if contains_null_cols:
+                    batch = pyarrow.remove_null_columns(batch)
```

Comment:
`remove_null_columns` should be done in `normalize_py_arrow_item`.
thanks to that we'll do it already in the extractor (which uses the same normalize function). this is how it was supposed to work.

in the extractor:
1. we extract dlt schema first so we'll include incomplete table on which we'll warn in normalizer
2. then we'll drop columns before we write arrow in the extractor
3. then in normalizer all is normalized and we do not need to rewrite